### PR TITLE
access log: add flag for "upstream retry count exceeded"

### PR DESCRIPTION
**Pull request by ggreenway**
Signed-off-by: Greg Greenway <ggreenway@apple.com>

*Risk Level*: Low
*Testing*: New unit tests
*Docs Changes*: Added
*Release Notes*: Added

### CODE REVIEW

File: docs/root/configuration/access_log.rst [hunk 1]
```diff
@@ -205,6 +206,8 @@ The following command operators are supported:
     * **DI**: The request processing was delayed for a period specified via :ref:`fault injection <config_http_filters_fault_injection>`.
     * **FI**: The request was aborted with a response code specified via :ref:`fault injection <config_http_filters_fault_injection>`.
     * **RL**: The request was ratelimited locally by the :ref:`HTTP rate limit filter <config_http_filters_rate_limit>` in addition to 429 response code.
+    * **UAEX**: The request was denied by the external authorization service.
+    * **RLSE**: The request was rejected because there was an error in rate limit service.
```
Comment (dnoe) [inline docs/root/configuration/access_log.rst]: I assume these were previous missed and you discovered them when you were doing this, thanks for fixing.

File: api/envoy/data/accesslog/v2/accesslog.proto [hunk 2]
```diff
@@ -185,12 +185,14 @@ message ResponseFlags {
   // Indicates if the request was deemed unauthorized and the reason for it.
   Unauthorized unauthorized_details = 13;
 
-  // [#not-implemented-hide:] Hide from docs.
```
Comment (dnoe) [inline api/envoy/data/accesslog/v2/accesslog.proto]: Unrelated clean up?

File: docs/root/configuration/access_log.rst (hunk 1 above)
Comment (ggreenway) [reply docs/root/configuration/access_log.rst]: Yep

File: api/envoy/data/accesslog/v2/accesslog.proto (hunk 2 above)
Comment (ggreenway) [reply api/envoy/data/accesslog/v2/accesslog.proto]: I discovered that it was in fact implemented already, so I figured I'd fix the docs while I was touching everything:

https://github.com/envoyproxy/envoy/blob/b71bd22c99d36037e317a65260eba75a91cdafd7/source/extensions/access_loggers/http_grpc/grpc_access_log_impl.cc#L148

**Pull request merged**